spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java

```diff
@@ -104,8 +104,10 @@ Mono<String> empty() {
 
             private static void addListenerForCountingCompletedRequests() {
                 ServiceRequestContext.current().log().whenComplete()
-                                     .thenAccept(log -> completed.incrementAndGet());
-                requestReceived.set(true);
+                                     .thenAccept(log -> {
+                                         completed.incrementAndGet();
+                                         requestReceived.set(true);
+                                     });
```

Comment:
> https://github.com/line/armeria/pull/4727#issuecomment-1567827284
![image](https://github.com/line/armeria/assets/34854527/25c5cbb5-be11-4ae5-972a-1de6a03d1915)

Not related to this PR, I think there's some timing issue on `ByteBufLeakTest`. (checking by my local env and unit test)

```[h1c://ad01742424.local/mono#GET] Response stream has been cancelled.
com.linecorp.armeria.common.stream.CancelledSubscriptionException: null
	at com.linecorp.armeria.common.stream.CancelledSubscriptionException.get(CancelledSubscriptionException.java:40)
	at com.linecorp.armeria.common.stream.PublisherBasedStreamMessage$AbortableSubscriber.cancelOrAbort0(PublisherBasedStreamMessage.java:278)

refCnt: 1

org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a com.linecorp.armeria.spring.web.reactive.ByteBufLeakTest 
expected: 0
 but was: 1 within 10 seconds.
Expected :0
Actual   :1
```
`requestReceived.set(true);` should called on `thenAccept`, after each request is completed? 🤔

Reply:
```java
// ByteBufLeakTest#confirmNoBufferLeak_resetConnection test
        for (int i = 0; i < 2 * 3; i++) {
            try (Socket s = new Socket(NetUtil.LOCALHOST, port)) {
                requestReceived.set(false);
                s.setSoLinger(true, 0);
                final PrintWriter outWriter = new PrintWriter(s.getOutputStream(), false);
                switch (i % 3) {
                    case 0:
                        outWriter.print("GET /mono HTTP/1.1\r\n\r\n");
                        break;
                    case 1:
                        outWriter.print("GET /flux HTTP/1.1\r\n\r\n");
                        break;
                    case 2:
                        outWriter.print("GET /empty HTTP/1.1\r\n\r\n");
                        break;
                }
                outWriter.flush();

                // requestReceived can be set=true although request is not completed yet ..
                await().untilTrue(requestReceived); // here! currently we can goto next outWrtier.print although request is not completed.
            }
        }
```